core/src/org/labkey/core/admin/AdminController.java

```diff
@@ -10952,10 +10952,16 @@ public Object execute(SimpleApiJsonForm form, BindException errors) throws Excep
             {
                 var jsonStr = jsonObj.toString();
                 JSONObject cspReport = jsonObj.getJSONObject("csp-report");
-                String urlString = cspReport.getString("source-file");
-                String path = new URLHelper(urlString).deleteParameters().getPath();
-                if (null == reports.put(path,Boolean.TRUE))
-                    _log.warn("ContentSecurityPolicy warning:\n" + jsonStr);
+                if (cspReport != null)
+                {
+                    String urlString = cspReport.getString("document-uri");
+                    if (urlString != null)
+                    {
+                        String path = new URLHelper(urlString).deleteParameters().getPath();
+                        if (null == reports.put(path, Boolean.TRUE))
+                            _log.warn("ContentSecurityPolicy warning on page: " + urlString + "\n" + jsonStr);
+                    }
+                }

```

Comment:
Should use `JSONObject.opt*` for items that might not be there; `get*` will just throw an exception if they don't find anything. Also, `JSONObject.toString` can take an indent factor to make it look pretty.
```suggestion
                JSONObject cspReport = jsonObj.optJSONObject("csp-report");
                if (cspReport != null)
                {
                    String urlString = cspReport.optString("document-uri", null);
                    if (urlString != null)
                    {
                        String path = new URLHelper(urlString).deleteParameters().getPath();
                        if (null == reports.put(path, Boolean.TRUE))
                        {
                            var jsonStr = jsonObj.toString(2);
                            _log.warn("ContentSecurityPolicy warning on page: " + urlString + "\n" + jsonStr);
                        }
                    }
                }
```